tests/src/bin/analyze.rs

```diff
@@ -218,3 +232,125 @@ fn fmt_and_write_svg(path: impl AsRef<std::path::Path>, svg: &str) -> Result<()>
     std::fs::write(&path, out.as_ref())?;
     Ok(())
 }
+
+struct CurveId {
+    curve: String,
+    operation: String,
+}
+impl std::str::FromStr for CurveId {
+    type Err = anyhow::Error;
+    fn from_str(id: &str) -> std::prelude::v1::Result<Self, Self::Err> {
+        let regex = regex::Regex::new(r"^([^/]+?)/(.+?)$").context("construct regex")?;
+        let captures = regex.captures(id).context("id doesn't match regex")?;
+        let (_, [curve, operation]) = captures.extract();
+
+        Ok(Self {
+            curve: curve.to_owned(),
+            operation: operation.to_owned(),
+        })
+    }
+}
+
+fn analyze_curves_perf() -> Result<()> {
+    let stdin = std::io::stdin().lock();
+    let stdin = serde_json::de::IoRead::new(stdin);
+
+    let results = parse_completed_benchmarks(stdin).context("parse results")?;
+    let results = results
+        .into_iter()
+        .map(|res| {
+            Ok(BenchmarkComplete {
+                id: res.id.parse::<CurveId>()?,
+                mean: res.mean,
+            })
+        })
+        .collect::<Result<Vec<_>>>()?;
+
+    let mut grouped_results: Vec<(String, Vec<(String, Measurement)>)> = vec![];
+    for res in results {
+        let out = match grouped_results.iter_mut().find(|r| r.0 == res.id.operation) {
+            Some(o) => o,
+            None => {
+                grouped_results.push((res.id.operation.clone(), vec![]));
+                grouped_results.last_mut().unwrap()
+            }
+        };
+        out.1.push((res.id.curve.clone(), res.mean.clone()))
+    }
+
+    let curves = grouped_results[0]
+        .1
+        .iter()
+        .map(|(curve, _mean)| curve.clone())
+        .collect::<Vec<_>>();
+
+    let mut table = tabled::builder::Builder::new();
+    table.push_record(iter::once("").chain(curves.iter().map(String::as_str)));
+
+    for res in &grouped_results {
+        let operation = res.0.clone().replace('[', "\\[").replace(']', "\\]");
+        let mut row = vec![operation];
+        row.extend(iter::repeat_with(|| String::new()).take(curves.len()));
+
+        let (means, unit) = choose_uniform_unit(res.1.iter().map(|(_, m)| m));
+        for (curve, mean) in res.1.iter().map(|(curve, _)| curve).zip(means) {
+            let pos = curves.iter().position(|c| c == curve).unwrap();
+            row[1 + pos] = if unit == "ns" {
+                format!("{mean:.0}{unit}")
+            } else {
+                format!("{mean:.1}{unit}")
+            };
+        }
+
+        table.push_record(row);
+    }
+
+    println!(
+        "{}",
+        table
+            .build()
+            .with(tabled::settings::style::Style::markdown())
+    );
+
+    Ok(())
+}
+
+fn choose_uniform_unit<'a>(
+    measurements: impl Iterator<Item = &'a Measurement> + Clone,
+) -> (Vec<f64>, &'static str) {
```

Comment:
Why not map back to measurements, only adjusting their units?

Reply:
In this way, signature of function guarantees that units are indeed the same

Reply:
Alternatively, since measurements are no longer displayed, the Display instance above is not necessary?

Reply:
Yep I just deleted it (it was buggy anyways)